imod/msw/meteo_mapping.py

```diff
@@ -1,18 +1,43 @@
 from copy import deepcopy
-from typing import Any, Optional, TextIO
+from pathlib import Path
+from typing import Any, Optional, TextIO, cast
 
 import numpy as np
 import pandas as pd
 import xarray as xr
 
+import imod
 from imod.mf6.utilities.regrid import RegridderWeightsCache
 from imod.msw.fixed_format import VariableMetaData
 from imod.msw.pkgbase import MetaSwapPackage
+from imod.msw.utilities.common import find_in_file_list
 from imod.prepare import common
-from imod.typing import GridDataArray, IntArray
+from imod.typing import GridDataArray, Imod5DataDict, IntArray
 from imod.util.regrid_method_type import RegridMethodType
 
 
+def open_first_meteo_grid(mete_grid_path: str | Path, column_nr: int) -> xr.DataArray:
+    """
+    Find and open first meteo grid path in mete_grid.inp. This grid is enough to
+    generate meteomappings.
+    """
+    if column_nr not in [2, 3]:
+        raise ValueError("Column nr should be 2 or 3")
+
+    mete_grid_path = Path(mete_grid_path)
+
+    with open(mete_grid_path, "r") as f:
+        first_line = f.readline()
```

Comment:
MetaSWAP also supports constants as input (I found out recently). According to the manual there should be at least one meteo-file present. I would suggest a open_first_present_meteo_grid method.

Reply:
Implemented!